### PR TITLE
Add license explanation issue

### LINKED ISSUE #292: Add license explanation tool tip in the filter section
The users can click the question mark icons in the filter section on [CC Search](https://search.creativecommons.org/search?q=mountain) to see a brief explainer of each license. 

Screenshot from 2020-10-01 22-42-25

The same is to be implemented in the filters section of the extension. 

- The icons are already available in the [vocabulary](https://cc-vocabulary.netlify.app/?path=/docs/tokens-icons--figma)
- The design and the content will be identical to the CC Search implementation.

### PR DESCRIPTION
**Pull request by patelaryan7751**
Fixes #292 

**Description**

issue is fixed added all explanation of licenses and ran it locally you can find the gif of it below
dwork

 

**Other information**

**Checklist:**
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `master` branch of the repository.
- [x] My commit messages follow [best practices][best_practices].
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

### CODE REVIEW

**Comment by makkoncept** [comment]
@patelaryan7751 This is in the context of your message on slack.

You can see in [the logs](https://github.com/creativecommons/ccsearch-browser-extension/pull/304/checks?check_run_id=1361589596) that the formatting in `popup.html` is the cause of failing checks.

I would recommend you to read the code formatting section in [INSTRUCTIONS.md](https://github.com/creativecommons/ccsearch-browser-extension/blob/master/INSTRUCTIONS.md#using-command-line) to see how you can tackle this.

Also, please push the updates in the same PR and do not open new ones. You have already closed 9 PRs for the same issue.

**Comment by patelaryan7751** [comment]
> @patelaryan7751 This is in the context of your message on slack.
> 
> You can see in [the logs](https://github.com/creativecommons/ccsearch-browser-extension/pull/304/checks?check_run_id=1361589596) that the formatting in `popup.html` is the cause of failing checks.
> 
> I would recommend you to read the code formatting section in [INSTRUCTIONS.md](https://github.com/creativecommons/ccsearch-browser-extension/blob/master/INSTRUCTIONS.md#using-command-line) to see how you can tackle this.
> 
> Also, please push the updates in the same PR and do not open new ones. You have already closed 9 PRs for the same issue.

hello @makkoncept  i  have followed the guidelines and all the checks passed .

**Review by makkoncept — CHANGES_REQUESTED** [review]
The design of the tool-tip is not one that we want. As already mentioned in the corresponding issue, _The design and the content will be identical to the CC Search implementation._

Pasting the screenshot again for reference: 

Screenshot from 2020-10-01 22-42-25

**Comment by patelaryan7751** [comment]
> The design of the tool-tip is not one that we want. As already mentioned in the corresponding issue, _The design and the content will be identical to the CC Search implementation._
> 
> Pasting the screenshot again for reference:
> 
> Screenshot from 2020-10-01 22-42-25

@makkoncept  i would implement the same design . I am working on it . I guess rest of the functionality is working perfectly.

File: src/popup/helper.js [hunk 1]
```diff
@@ -112,3 +112,38 @@ export const licenseInfo = {
...
+    tooltipHeading: `License CC BY-NC`,
+    tooltipContent: `<p><i class='icon cc-by'></i> Credit the creator.<br/><i class='icon cc-nc'></i> Noncommercial uses only.</p>`,
+  },
+  bysa: {
+    tooltipHeading: `License CC BY-SA`,
+    tooltipContent: `<p><i class='icon cc-by'></i> Credit the creator.<br/><i class='icon cc-sa'></i> Share adaptations under the same terms.</p>`,
+  },
+  bynd: {
+    tooltipHeading: `License CC BY-ND`,
+    tooltipContent: `<p><i class='icon cc-by'></i> Credit the creator.<br/><i class='icon cc-nd'></i> No derivatives or adaptations permitted.</p>`,
```
Comment (makkoncept) [inline src/popup/helper.js:131]: As already said multiple times, the information is already present in `licenseInfo` object. There is no need to create a new one.

**Review by makkoncept — CHANGES_REQUESTED** [review]

File: src/popup/helper.js (hunk 1 above)
Comment (patelaryan7751) [reply src/popup/helper.js:131]: i would be working on the things as mentioned.

**Comment by patelaryan7751** [comment]
I think the below design would do good.
dwork

i have used licenseInfo from helper.js in tooltipInfo to get the icons and their information dynamically .

**Comment by patelaryan7751** [comment]
I think the below design would do good.

i have used licenseInfo from helper.js in tooltipInfo to get the icons and their information dynamically .
i have created a new "tooltipInfo " because the heading of the licenses are not present in "licenseInfo". 
also there are license info which has multiple symbols like" CC-BY-SA" it has mutiple icons and multiple description for this "licenseInfo" may not be sufficient (as far as license heading is concerned) hence there was a need for creating "tooltipInfo".
if we can do it better then let me know i would surely work on it.
dwork

**Pull request closed without merging**